init.cc

```diff
@@ -39,7 +39,14 @@ std::set<gms::inet_address> get_seeds_from_db_config(const db::config& cfg, gms:
         }
     }
     if (seeds.empty()) {
-        seeds.emplace(gms::inet_address("127.0.0.1"));
+        auto localhost = gms::inet_address("127.0.0.1");
+        // Single node cluster?
+        if (cfg.developer_mode() || (listen == localhost && broadcast_address == localhost)) {
+            seeds.emplace(localhost);
```

Comment:
Does the developer_mode branch make sense? Let's say listen_address = broadcast_address = 1.2.3.4. Won't attempts to contact localhost fail?

Reply:
I guess, it's not a regression compared to current state, since current state unconditionally sets the seeds to localhost if they were empty.

Reply:
Right, and in developer mode there could be some virtual interface or iptables rules that we're not aware of.